lib/modules/datasource/index.ts

```diff
@@ -209,7 +209,9 @@ function resolveRegistryUrls(
         'Custom registries are not allowed for this datasource and will be ignored'
       );
     }
-    return datasource.defaultRegistryUrls ?? [];
+    return is.function_(datasource.defaultRegistryUrls)
+      ? datasource.defaultRegistryUrls()
```

Comment:
Can we ignore this coverage issue for now and handle it when we use the function?

Reply:
nope

Reply:
```js
@@ -17,7 +19,8 @@ export class GitlabTagsDatasource extends Datasource {
:...skipping...
diff --git a/lib/modules/datasource/gitlab-tags/index.ts b/lib/modules/datasource/gitlab-tags/index.ts
index bb9185f7a..8deb17b30 100644
--- a/lib/modules/datasource/gitlab-tags/index.ts
+++ b/lib/modules/datasource/gitlab-tags/index.ts
@@ -1,3 +1,4 @@
+import { GlobalConfig } from '../../../config/global';
 import { logger } from '../../../logger';
 import { cache } from '../../../util/cache/package/decorator';
 import { GitlabHttp } from '../../../util/http/gitlab';
@@ -7,6 +8,7 @@ import type { DigestConfig, GetReleasesConfig, ReleaseResult } from '../types';       
 import type { GitlabCommit, GitlabTag } from './types';
 import { defaultRegistryUrl, getDepHost, getSourceUrl } from './util';

+const { platform, endpoint } = GlobalConfig.get();
 export class GitlabTagsDatasource extends Datasource {
   static readonly id = 'gitlab-tags';

@@ -17,7 +19,8 @@ export class GitlabTagsDatasource extends Datasource {
     this.http = new GitlabHttp(GitlabTagsDatasource.id);
   }

-  override readonly defaultRegistryUrls = [defaultRegistryUrl];
+  override readonly defaultRegistryUrls =
+    platform === 'gitlab' && endpoint ? () => [endpoint] : [defaultRegistryUrl];

   @cache({
     namespace: `datasource-${GitlabTagsDatasource.id}`,
```

Reply:
I think we might need to call `GlobalConfig.get()` as part of `defaultRegistryUrls()`. Otherwise it will load it at the very start before GlobalConfig is initialized.